inc/functions_post.php

```diff
@@ -321,7 +321,10 @@ function build_postbit($post, $post_type=0)
 			eval("\$post['useravatar'] = \"".$templates->get("postbit_avatar")."\";");
 		}
 
-		eval("\$post['button_find'] = \"".$templates->get("postbit_find")."\";");
+		if($mybb->usergroup['cansearch'] == 1)
+		{
+			eval("\$post['button_find'] = \"".$templates->get("postbit_find")."\";");
```

Comment:
$post['button_find'] is now not defined when the user doesn't have the permission.